### DISCUSSION TITLE
plan.autoFenceOperations setting in parameters and per operation

### DISCUSSION THREAD

**Discussion by gibhsu** [frePPLe/frepple#480] [Q&A] [answered]
I would like to clarify the explanation for Release Fence on the following link.
https://frepple.org/docs/6.8/examples/operation/operation-autofence.php

The plan.autoFenceOperations in parameters setting is the default value for each operation if I endable the FENCE in the constraint planning.
The fence value in the operation will overwrite the default value if it is maintained. 
Is my understanding correct? Or they are different thing?

**Comment by hichamlahlou** [comment, marked as answer]
These are 2 different things:

The plan.autoFenceOperations is to let the solver know how long it can wait for existing confirmed supply before proposing another replenishment. Example: You have a PO due in 10 days. If plan.autoFenceOperations is 14, the solver will not propose another PO and wait for that one (provided its quantity covers the need). If plan.autoFenceOperations is 7, then another PO will be proposed.

The fence constraint is to respect the release fence field at operation level (operation + item supplier + item distribution). The solver will not propose any PO/MO/DO within the release fence. There is no default parameter for the release fence, it has to be set for each record of operation/item supplier/item distribution.
